src/dev/impl/DevToys/UI/Controls/CodeEditor.xaml.cs

```diff
@@ -505,6 +536,36 @@ private void ExpandButton_Click(object _, RoutedEventArgs e)
                 GetExpandButtonIcon().Glyph = "\uF15F";
                 ToolTipService.SetToolTip(GetExpandButton(), LanguageManager.Instance.Common.Expand);
             }
+
+            // Must dispatch this final resize to give the main control a chance to size properly
+            Window.Current.Dispatcher.RunIdleAsync(_ =>
```

Comment:
Is waiting for idle necessary? Perhaps https://github.com/veler/DevToys/blob/5b82435792870f193d8149e6d6b811e2d48a841c/src/dev/impl/DevToys/Core/Threading/ThreadHelper.cs#L37C30-L37C48 is enough, although I'd expect we already run on the UI thread here.

Reply:
> Is waiting for idle necessary?

Unfortunately I think so, because we need to wait until after the `CodeEditor`'s size has evaluated.

I tried `ThreadHelper.RunOnUIThreadAsync`, but as you said, we're already on the UI thread. I also tried `RunAsync` with every `CoreDispatcherPriority`, and none of them worked (actually `Idle` threw an exception).

I realize this is hacky, but it's the only way I could get the sizing to work. I can upload a video of the problem without this if you'd like.

Reply:
I see.
How about making `ShowInfoContent` a DependencyProperty and bind it to the Visibility property of `CodeEditorCoreContainer` and `InfoGrid` in the XAML by using a [BooleanToVisibilityConverter](https://github.com/veler/DevToys/blob/5b82435792870f193d8149e6d6b811e2d48a841c/src/dev/impl/DevToys/UI/Converters/BooleanToVisibilityConverter.cs#L12)?

I'm not sure to follow the sizing issue here otherwise.

Reply:
Good idea, binding the `Visibility` of both controls is a better approach and reduces the code-behind. I pushed this in d20981fb290ff6d3711cbd96cb744260015b0576.

However, it doesn't change the need to dispatch the property change. 😞

Reply:
Thanks! Could we try one last thing to replace RunIdleAsync? How about:

```
ShowInfoContent = showInfo;
UpdateLayout();
```

This should trigger the UI to rearrange et remeasure itself, I believe.

Reply:
Unfortunately this does not work either. I believe the problem is that the CodeEditor (whose sizing we are using as the source of truth) never gets the chance to be drawn since we change `ShowInfoContent` before the UI thread runs.

I will continue the conversation in your other comment.